contracts/upgradeable_contracts/multi_amb_erc20_to_erc677/HomeMultiAMBErc20ToErc677.sol

```diff
@@ -0,0 +1,254 @@
+    * @param _owner address of the owner of the mediator contract.
+    * @param _tokenImage address of the PermittableToken contract that will be used for deploying of new tokens.
+    */
+    function initialize(
+        address _bridgeContract,
+        address _mediatorContract,
+        uint256[3] _dailyLimitMaxPerTxMinPerTxArray, // [ 0 = _dailyLimit, 1 = _maxPerTx, 2 = _minPerTx ]
+        uint256[2] _executionDailyLimitExecutionMaxPerTxArray, // [ 0 = _executionDailyLimit, 1 = _executionMaxPerTx ]
+        uint256 _requestGasLimit,
+        address _owner,
+        address _tokenImage
+    ) external onlyRelevantSender returns (bool) {
+        require(!isInitialized());
+        require(_owner != address(0));
+
+        _setBridgeContract(_bridgeContract);
+        _setMediatorContractOnOtherSide(_mediatorContract);
+        _setLimits(address(0), _dailyLimitMaxPerTxMinPerTxArray);
+        _setExecutionLimits(address(0), _executionDailyLimitExecutionMaxPerTxArray);
+        _setRequestGasLimit(_requestGasLimit);
+        setOwner(_owner);
+        _setTokenImage(_tokenImage);
+        setInitialize();
+
+        return isInitialized();
+    }
+
+    /**
+    * @dev Updates an address of the token image contract used for proxifying newly created tokens.
+    * @param _tokenImage address of PermittableToken contract.
+    */
+    function setTokenImage(address _tokenImage) external onlyOwner {
+        _setTokenImage(_tokenImage);
+    }
+
+    /**
+    * @dev Retrieves address of the token image contract.
+    * @return address of block reward contract.
+    */
+    function tokenImage() public view returns (address) {
+        return addressStorage[TOKEN_IMAGE_CONTRACT];
+    }
+
+    /**
+    * @dev Handles the bridged tokens for the first time, includes deployment of new TokenProxy contract.
+    * Checks that the value is inside the execution limits and invokes the method
+    * to execute the Mint or Unlock accordingly.
+    * @param _token address of the bridged ERC20/ERC677 token on the foreign side.
+    * @param _name name of the bridged token, "x" will be appended, if empty, symbol will be used instead.
+    * @param _symbol symbol of the bridged token, "x" will be appended, if empty, name will be used instead.
+    * @param _decimals decimals of the bridge foreign token.
+    * @param _recipient address that will receive the tokens.
+    * @param _value amount of tokens to be received.
+    */
+    function deployAndHandleBridgedTokens(
+        address _token,
+        string _name,
+        string _symbol,
+        uint8 _decimals,
+        address _recipient,
+        uint256 _value
+    ) external {
+        string memory name = _name;
+        string memory symbol = _symbol;
+        if (bytes(name).length == 0) {
+            name = symbol;
+        } else if (bytes(symbol).length == 0) {
+            symbol = name;
+        }
+        name = string(abi.encodePacked("x", name));
```

Comment:
After discussion with Igor, we agreed to not add "x" at the beginning but add "on xDai" at the end.

Reply:
Addressed in 7865fb2 as well